mantidproject/mantid
# Re #21134: Fixed incorrect title when transfering runs with th in their name.

#### Description Of Work ####
- Minor refactoring on `ReflLegacyTransferStrategy`
- Uses additional capturing groups to capture text before and after the theta value eliminating the need for the `find("th")` hack.
- Removes support for `in {angle} theta` style descriptions.
- Switches to `std::regex` over `boost::regex` now that named capture groups are no longer required.

**To test:**
1. Open the reflectometry GUI.
2. Set the instrument to `OFFSPEC`
3. Enter the investigation id: `1720380` and click search.
4. Transfer all the runs into the table and ensure that: 
    - The the titles match the descriptions with the `th=..` trimmed.
    - The the runs where the descriptions only differ by their theta value have been grouped with the run numbers separated by a `+`.
5. Follow the instructions in #19260 and ensure that the bug is no longer present.
6. Code review



Fixes #21134 and fixes #19260. 

**Release Notes** 
*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
